src/components/Search/Search.tsx

```diff
@@ -1,181 +1,195 @@
+                      break;
+                    default:
+                      break;
+                  }
+                }
+              }
+            },
+          )
+          .catch((error: string) => error);
+      });
+    }
+  }, []);
+
+  const gotoClassDate = (date: Date) => {
+    setActiveDay(date);
+  };
+
+  const formattedDate = (date:any) => {
+    const year = date.getFullYear();
+    const month = String(date.getMonth() + 1).padStart(2, '0');
+    const day = String(date.getDate()).padStart(2, '0');
+    return `${year}-${month}-${day}`;
+  };
+
+  const search = async () => {
+    const searchByDate = formattedDate(activeDay);
+    setClasses([]);
+    setSearched(true);
+    setList(true);
+
+    const zipRegex = /^[0-9]{5}(?:-[0-9]{4})?$/;
+    let column = '';
+    if (zipRegex.test(searchByLocation)) {
+      column = 'zip';
+    } else {
+      column = 'city';
+    }
+    try {
+      const { data: locations, error } = await supabase
+        .from('locations')
+        .select('*')
+        .ilike(column, searchByLocation);
+      if (error) {
+        return error;
+      }
+      locations.forEach(async (location) => {
+        const { data: classes } = await supabase
+          .from('classes')
+          .select('*, locations(*)')
+          .eq('location_id', location.location_id)
+          .eq('date', searchByDate)
+          .order('time', { ascending: false });
+        if (classes) {
+          classes.forEach((Class) => {
+            if (!searchByClass
+            || (searchByClass && Class.name.toLowerCase().includes(searchByClass.toLowerCase()))) {
+              setClasses((prevClasses) => [...prevClasses, Class]);
+            }
+          });
+        }
+      });
+    } catch (err) {
+      console.error('Unexpected error: ', err);
+    }
+    return null;
+  };
+
+  // for testing suite
+  onSearch();
+  //
+
+  let classLabel: JSX.Element | null = null;
+  if (searched && Classes.length <= 1) {
+    classLabel = (
+      <div>
+        {Classes.length}
+        {' '}
+        class
+      </div>
+    );
+  } else if (searched && Classes.length > 1) {
+    classLabel = (
+      <div>
+        {Classes.length}
+        {' '}
+        classes
+      </div>
+    );
+  }
+
+  return (
+    <div className="text-black mt-10">
+      <div className="flex">
+        <DatePicker
+          activeDay={activeDay}
+          gotoClassDate={gotoClassDate}
+        />
+        <input
+          data-testid="search-input"
+          className="w-[140px] text-center font-normal text-xs rounded-s-md"
+          placeholder="Yoga, Pilates, Zumba..."
+          onChange={(e) => setSearchByClass(e.target.value)}
+        />
+        <div className="flex items-center text-white font-normal bg-seafoam"><MapPin className="h-5 w-5" /></div>
+        <input
+          className="w-[100px] text-center font-normal text-xs"
+          placeholder={searchByLocation}
```

Comment:
This placeholder should be an example similar to above, it defaults to an empty string so it is not clear what to enter into this box

Reply:
updated, make placeholder = "City, location..."